src/storage/src/db_store.rs

```diff
@@ -181,23 +182,91 @@ impl DbStore {
         Ok(())
     }
 
+    //
+    // add document
+    //
+    fn add_document(
+        db: Pin<&mut Merk>,
+        sender: &DB3Address,
+        tx: &TxId,
+        mutation: &DatabaseMutation,
+        blockId: u64,
+        mutationId: u32,
+    ) -> Result<()> {
+        let addr_ref: &[u8] = mutation.db_address.as_ref();
+        let db_id = DbId::try_from(addr_ref)?;
+        let database = Self::get_database(db.as_ref(), &db_id)?;
+        match database {
+            Some(d) => {
+                let mut entries: Vec<BatchEntry> = Vec::new();
+                let mut idx = 0;
```

Comment:
let cid_set = HashSet::from(d.collections.iter().map(|c| c.name.to_string()));

Reply:
Create a cid_index_map to enable retrieval of index_list using `collection_name`.